plugin/logger.go

```diff
@@ -0,0 +1,83 @@
+package plugin
+
+import (
+	"encoding/json"
+	"fmt"
+
+	logging "github.com/ipfs/go-log"
+
+	"github.com/ipfs/go-ipfs/plugin"
+)
+
+var _ plugin.Plugin = &LoggerPlugin{}
+
+// "Plugins": {
+//      "logger": {
+//        "Config": {
+//            "Subsystems": ["dht","relay","corerepo"],
```

Comment:
I don't know if that's the most useful UX. What about:

```
levels: {
    fatal: [system1, system2, ... ]
    error: [...]
    info: [...]
    ...
},
defaultLevel: "info",
```

Reply:
This way we don't have to enumerate every system and if new one are added they don't get dropped.